### DISCUSSION TITLE
TextEdit cursor blink

### DISCUSSION THREAD

**Discussion by mlrv** [emilk/egui#2349] [Q&A]
I can't find any native way to set up the cursor of a TextEdit to blink (like in a normal web-based text area). The only thing I managed to achieve was to simulate blinking by periodically changing the width of the cursor from 1.0 to 0.0 - this "works", but has the downside of requiring continuous mode (I might be wrong here, please correct me!), which in turns makes the app quite laggy and a lot less performant.

```rust
fn update(&mut self, ctx: &Context, _frame: &mut Frame) {
  // ...
  // Used to simulate cursor blinking
  ctx.request_repaint();

  let show_cursor = (ui.input().time * 4.0).floor() as i64 % 4 != 0;
  style.visuals.text_cursor_width = if show_cursor { 1.0 } else { 0.0 };

  ctx.set_style(style);
  // ...
}
```

Is there another way to do this? Or, alternatively, is there a way to make this performant?

**Comment by n00kii** [comment]
have you considered doing this from a different thread, sleeping the thread for the delay period and only requesting a repaint as needed?

**Reply by mlrv** [reply]
I'm very new to Rust, so multi-threading is not something I've approached yet 😄  but this definitely sounds like a possibility!
